src/CompiledSouffle.h

```diff
@@ -234,4 +234,92 @@ class t_nullaries {
     void printHintStatistics(std::ostream& o, std::string prefix) const {}
 };
 
+/** info relations */
+template <int Arity>
+class t_info {
+private:
+    std::vector<ram::Tuple<RamDomain, Arity>> data;
+    Lock insert_lock;
+
+public:
+    t_info() = default;
+    using t_tuple = ram::Tuple<RamDomain, Arity>;
+    struct context {};
+    context createContext() {
+        return context();
+    }
+    class iterator : public std::iterator<std::forward_iterator_tag, ram::Tuple<RamDomain, Arity>> {
+        typename std::vector<ram::Tuple<RamDomain, Arity>>::const_iterator it;
+
+    public:
+        iterator(const typename std::vector<t_tuple>::const_iterator& o) : it(o) {}
+
+        const t_tuple operator*() {
+            return *it;
+        }
+
+        bool operator==(const iterator& other) const {
+            return other.it == it || *other.it == *it;
+        }
+
+        bool operator!=(const iterator& other) const {
+            return !(*this == other);
+        }
+
+        iterator& operator++() {
+            it++;
+            return *this;
+        }
+    };
+    iterator begin() const {
+        return iterator(data.begin());
+    }
+    iterator end() const {
+        return iterator(data.end());
+    }
+    void insert(const t_tuple& t) {
+        insert_lock.lock();
+        if (!contains(t)) {
+            data.push_back(t);
+        }
+        insert_lock.unlock();
+    }
+    void insert(const t_tuple& t, context& /* ctxt */) {
+        insert(t);
+    }
+    void insert(const RamDomain* ramDomain) {
+        insert_lock.lock();
+        t_tuple t;
+        for (size_t i = 0; i < Arity; ++i) {
+            t.data[i] = ramDomain[i];
+        }
+        data.push_back(t);
+        insert_lock.unlock();
+    }
+    bool insert() {
```

Comment:
I don't think we need this method. This is a legacy of the nullary class?

Reply:
It is legacy of nullary relation. I ll delete it